mlflow/gateway/providers/openai.py

```diff
@@ -0,0 +1,151 @@
+        #          },
+        #          "finish_reason":"stop",
+        #          "index":0
+        #       }
+        #    ]
+        # }
+        # ```
+        return chat.ResponsePayload(
+            **{
+                "candidates": [
+                    {
+                        "message": {
+                            "role": c["message"]["role"],
+                            "content": c["message"]["content"],
+                        },
+                        "metadata": {
+                            "finish_reason": c["finish_reason"],
+                        },
+                    }
+                    for c in resp["choices"]
+                ],
+                "metadata": {
+                    "input_tokens": resp["usage"]["prompt_tokens"],
+                    "output_tokens": resp["usage"]["completion_tokens"],
+                    "total_tokens": resp["usage"]["total_tokens"],
+                    "model": resp["model"],
+                    "route_type": self.config.type,
+                },
+            }
+        )
+
+    async def completions(self, payload: completions.RequestPayload) -> completions.ResponsePayload:
+        resp = await openai.Completion.acreate(
+            model=self.config.model.name,
+            prompt=payload.prompt,
+        )
+        # Response example (https://platform.openai.com/docs/api-reference/completions/create)
+        # ```
+        # {
+        #   "id": "cmpl-uqkvlQyYK7bGYrRHQ0eXlWi7",
+        #   "object": "text_completion",
+        #   "created": 1589478378,
+        #   "model": "text-davinci-003",
+        #   "choices": [
+        #     {
+        #       "text": "\n\nThis is indeed a test",
+        #       "index": 0,
+        #       "logprobs": null,
+        #       "finish_reason": "length"
+        #     }
+        #   ],
+        #   "usage": {
+        #     "prompt_tokens": 5,
+        #     "completion_tokens": 7,
+        #     "total_tokens": 12
+        #   }
+        # }
+        # ```
+        return completions.ResponsePayload(
+            **{
+                "candidates": [
+                    {
+                        "text": c["text"],
+                        "metadata": {"finish_reason": c["finish_reason"]},
+                    }
+                    for c in resp["choices"]
+                ],
+                "metadata": {
+                    "input_tokens": resp["usage"]["prompt_tokens"],
+                    "output_tokens": resp["usage"]["completion_tokens"],
+                    "total_tokens": resp["usage"]["total_tokens"],
+                    "model": resp["model"],
+                    "route_type": self.config.type,
+                },
+            }
+        )
+
+    async def embeddings(self, payload: embeddings.RequestPayload) -> embeddings.ResponsePayload:
+        resp = await openai.Embedding.acreate(
+            model=self.config.model.name,
+            input=payload.text,
```

Comment:
```suggestion
            input=payload.text,
            ...
```

Same here: https://github.com/mlflow/mlflow/pull/8697/files#r1227788703